napari/components/layerlist.py

```diff
@@ -62,6 +62,41 @@ class LayerList(SelectableEventedList[Layer]):
     selection.events._current : (value: _T)
         emitted when the current item has changed. (Private event)
 
+    Notes
+    -----
+
+    Note that ``changed`` events are only emitted when the value of an item in
+    the layer list item value is changed, not when the item is inserted or removed.
+
+    Examples
+    --------
+
+    >>> import napari
+    >>> from skimage.data import astronaut
+    >>> viewer = napari.Viewer()
+    >>> event_list = []
+
+    Connect to the event list:
+
+    >>> viewer.layers.events.connect(event_list.append)
+    <built-in method append of list object at 0x7fc225764780>
+
+    >>> viewer.add_image(astronaut())
+    <Image layer 'Image' at 0x7f0fe7fa9e90>
+    >>> viewer.add_points()
+    <Points layer 'Points' at 0x7f102f962350>
+    >>> viewer.layers
+    [<Image layer 'Image' at 0x7f0fe7fa9e90>, <Points layer 'Points' at 0x7f102f962350>]
+
+    Inspecting the list of events, we see:
+
+    >>> event_list[0].type
+    'inserting'
+    >>> viewer.layers.pop(1)
+    <Points layer 'Points' at 0x7f102f962350>
+    >>> event_list[-1].type
+    'removed'
```

Comment:
Did we have something that shows when `changed` is emitted?

Reply:
@Czaki see this comment on Zulip: https://napari.zulipchat.com/#narrow/channel/298358-working-group-documentation/topic/layerlist.20event.20type.20.22changed.22/near/484851070

you would need to do `viewer.layers[idx] = layer_instance`, which is not an API we want to encourage. This is because we inherit from EventedList — in the more general case, this would indeed be a useful event.